### PR TITLE
compiler: Cache use/def sets in alias analysis pass

### PR DESCRIPTION
**Pull request by frej**
Reduce the cost of the alias analysis pass by hoisting the calculation of use and def sets out of the liveness fixpoint iteration using a cache. On the modules compiled by `scripts/diffable` the time spent in the alias analysis pass is reduced by 16%.

### CODE REVIEW

File: lib/compiler/src/beam_ssa_alias.erl [hunk 1]
```diff
@@ -153,17 +154,28 @@ blk_effective_uses(#b_blk{is=Is,last=Last}) ->
...
-                   sets:del_element(Dst, Uses0);
-               _ -> % A terminator, no defs
-                   Uses0
-           end,
+                #b_set{dst=Dst} ->
+                    %% The uses after the def do not count
+                    sets:del_element(Dst, Uses0);
+                _ -> % A terminator, no defs
+                    Uses0
+            end,
```
Comment (bjorng) [inline lib/compiler/src/beam_ssa_alias.erl:162]: I see only whitespace changes on these lines. (And the indentation was correct in my Emacs before the extra whitespace was added.)

**Review by bjorng — COMMENTED** [review]
Looks good to me except for lines 157-161.

**Comment by bjorng** [comment]
I've removed the whitespace changes and force-pushed. Added to our daily builds.

File: lib/compiler/src/beam_ssa_alias.erl (hunk 1 above)
Comment (frej) [reply lib/compiler/src/beam_ssa_alias.erl]: Sloppy last minute revert of a change which did not have an impact on the performance, bad Frej! (I see that you already fixed it).

**Pull request merged**